### PR TITLE
Adding code viewer component

### PR DESCRIPTION
**Pull request by alexandreGrenierIRCC**

### CODE REVIEW

**Review by MikeJalow — APPROVED** [review]

File: component-library/src/app/components/accordion-panel/accordion-container.component.html
Comment (kristopher-houston) [inline component-library/src/app/components/accordion-panel/accordion-container.component.html:1]: extraLeft and extraRight?

File: component-library/src/app/components/code-viewer/code-viewer.component.html [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  [id]="config.id"
+  class="code-viewer"
+>
+  <ircc-cl-lib-tabs
+    class="tabs-component"
+    [config]="tabconfig"
+    (valueChange)="getSelectedTab($event)"
+  ></ircc-cl-lib-tabs>
+  <app-accordian-container [open]="openAccordion">
+    <ircc-cl-lib-button
```
Comment (kristopher-houston) [inline component-library/src/app/components/code-viewer/code-viewer.component.html:11]: Ah, I assume this allows you to put a button on one side or the other of the accordion.

File: component-library/src/app/components/code-viewer/code-viewer.component.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  SimpleChanges
+} from '@angular/core';
+import { TranslateService } from '@ngx-translate/core';
+import {
+  IButtonConfig,
+  ITabNavConfig
+} from 'ircc-ds-angular-component-library';
+import { SlugifyPipe } from 'src/app/share/pipe-slugify.pipe';
+
+export interface ICodeViewerTab {
```
Comment (kristopher-houston) [inline component-library/src/app/components/code-viewer/code-viewer.component.ts:16]: Is there any difference between this and ITabConfig?

File: component-library/src/app/components/code-viewer/code-viewer.component.ts [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    id: 'accordion-button',
+    category: 'plain',
+    size: 'small',
+    ariaLabel: 'Click to copy the code'
+  };
+
+  constructor(private translate: TranslateService) {}
+
+  ngOnInit() {
+    this.setupComponent();
```
Comment (kristopher-houston) [inline component-library/src/app/components/code-viewer/code-viewer.component.ts:63]: But... that's what ngOnInit is for...

File: component-library/src/app/components/code-viewer/code-viewer.component.ts [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+  height: number = 150;
+
+  buttonConfigCopy: IButtonConfig = {
+    id: 'accordion-button',
+    category: 'plain',
+    size: 'small',
+    ariaLabel: 'Click to copy the code'
+  };
+
+  constructor(private translate: TranslateService) {}
```
Comment (kristopher-houston) [inline component-library/src/app/components/code-viewer/code-viewer.component.ts:60]: Not actually used anywhere. Delete entire constructor.

File: component-library/src/app/pages/code-view/code-view.component.ts [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+  altLangLink = 'codeview'; // ROUTE translation path
+
+  codeViewConfig: ICodeViewerConfig = {
+    id: 'code-viewer-demo'
+  };
+
+  constructor(
+    private translate: TranslateService,
+    private lang: LangSwitchService,
+    private slugify: SlugifyPipe
```
Comment (kristopher-houston) [inline component-library/src/app/pages/code-view/code-view.component.ts:25]: Not used in the ts file. Remove.

File: component-library/src/app/components/accordion-panel/accordion-container.component.ts [hunk 6]
```diff
@@ -8,7 +8,8 @@ import { IButtonConfig } from 'ircc-ds-angular-component-library';
   styleUrls: ['./accordion-container.component.scss']
 })
 export class accordionContainerComponent {
```
[the remark this answers is not in the document]
Comment (alexandreGrenierIRCC) [reply component-library/src/app/components/accordion-panel/accordion-container.component.ts:10]: Added


export interface IAccordionContainerConfig {
  id: string;
  buttonText?: string;
  buttonTextClosed?: string;
  open?: boolean;
}

**Pull request merged**